app/javascript/src/Settings/components/Common/TextInputGroup.jsx

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import * as React from 'react'
+import { useState } from 'react'
+import { FormGroup, TextInput } from '@patternfly/react-core'
+import type { FormSettingsProps } from 'Settings/types'
+
+const TextInputGroup = (
+  {
+    defaultValue,
+    placeholder,
+    label,
+    name,
+    hint,
+    value,
+    isDefaultValue = false,
+    readOnly = false,
+    inputType = 'text'
+  }: FormSettingsProps) => {
+  const [ inputValue, setValue ] = useState(value)
+  const onChange = (value) => setValue(value)
```

Comment:
I think this is equivalent:
```suggestion
  const onChange = setValue
```

Reply:
It would be, but it's coded that way to avoid React warning about hooks are not prepared for callbacks. This happens because of PF `OnChange` function signature, being the second argument the `event`. I'll make it more explicit